### ISSUE TITLE
Test for WCAG success criterion 1.4.3: Contrast

### ISSUE THREAD

**Issue by hannolans** [quailjs/quail#186]
http://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html
The visual presentation of text and images of text has a contrast ratio of at least 4.5:1
## Applicability

Always
## Selector

Text parts:
- Text is less than 18 point if not bold and less than 14 point if bold (Situation A)
- Text is at least 18 point if not bold and at least 14 point if bold (Situation B)

No minimum contrast requirement for text or images of text:
- that are part of an inactive user interface component
- that are pure decoration
- that are not visible to anyone, 
- that are part of a picture that contains significant other visual content
- that is part of a logo or brand name
## Failures
- [ ] F24: Specifying foreground colors without specifying background colors or vice versa (#187 )
- [ ] F83: Using background images that do not provide sufficient contrast with foreground text (or images of text) 
## Techniques
- [ ] G148: Not specifying background color, not specifying text color, and not using technology features that change those defaults (see F24 #187)
- [ ] G174: Providing a control with a sufficient contrast ratio that allows users to switch to a presentation that uses sufficient contrast 
- [x] G18: Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text for situation A AND G145: Ensuring that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text for situation B ( #71 )
## Relationship between techniques and Failures
- G18 and G145 are exactly the same, only the ratio differs
- G18 is also used on AAA level, but then for situation B.
- F24 is counterpart for G148
- F83 is part of the check in G18+G145, but as images can't change their contrast with a control or backgorund css it's a real failure and G174 or G148 can't help that.
- G18 + G145 is the most common technology, but if they fail, we can't fail this success criterion if we don't check for G148 (easy) or G174 (difficult as we should detect a control)
